RELEASE-NOTES.rst

```diff
@@ -11,15 +11,26 @@ Invenio module that stores and registers persistent identifiers.
 
 *This is an experimental development preview release.*
 
-Improved features
------------------
+Incompatible changes
+--------------------
 
-- Makes Datacite provider optional.
+- Major rewrite of PIDStore in order to fix performance issues and
+  clean up programmatic API. (#21) (#20) (#19) (#9) (#4) (#1)
+- Refactoring and column name change for pidSTORE table. * Added unit
```

Comment:
Missing newline before `* Added unit`.  

However, do we want to start generating release notes already?  We have not been doing it for alpha pre-releases... Perhaps this can wait for beta.